bot/exts/moderation/incidents.py

```diff
@@ -173,18 +243,30 @@ class Incidents(Cog):
         * If `Signal.ACTIONED` or `Signal.NOT_ACTIONED` were chosen, attempt to
           relay the incident message to #incidents-archive
         * If relay successful, delete original message
+        * Search the cache for the webhook message for this message, if found delete it.
         * See: `on_raw_reaction_add`
 
     Please refer to function docstrings for implementation details.
     """
 
+    # This dictionary maps a incident message to the message link embeds sent by it
+    # RedisCache[discord.Message.id, str]
+    message_link_embeds_cache = RedisCache()
+
     def __init__(self, bot: Bot) -> None:
         """Prepare `event_lock` and schedule `crawl_task` on start-up."""
         self.bot = bot
 
+        self.bot.loop.create_task(self.get_webhook())
+
         self.event_lock = asyncio.Lock()
         self.crawl_task = self.bot.loop.create_task(self.crawl_incidents())
 
+    async def get_webhook(self) -> None:
+        """Fetch and store message link embeds webhook, present in #incidents channel."""
+        await self.bot.wait_until_guild_available()
+        self.incidents_webhook = await self.bot.fetch_webhook(Webhooks.incidents)
```

Comment:
If the webhook isn't available and returns `None`  this should be properly logged instead of letting some NPEs appear later on.

Reply:
Scaleios suggested doing something like this: https://discord.com/channels/267624335836053506/635950537262759947/833165964014256159.

Is there any better solution or should I go ahead and try doing this?

Reply:
Ok, I understood it wrong, will do the required changes.

Reply:
Was this addressed? doesn't seem so.

Reply:
Yeah I know that hence didn’t resolve it, had a [doubt](https://discord.com/channels/267624335836053506/635950537262759947/840543183674933270).

Reply:
Resolved in 9cfd4082.